### DISCUSSION TITLE
Fixed distance from top of the page / SILE internals documentation?

### DISCUSSION THREAD

**Discussion by raphCode** [sile-typesetter/sile#1628] [Q&A] [answered]
Hi, I would like to typeset the title page of my thesis, like this:
![image](https://user-images.githubusercontent.com/15750438/203958427-4a0fcfa9-5967-459d-9205-a57c18c6c0fb.png)

But I fail to add the vertical space at the beginning of the page.

I looked at the source and found this:
https://github.com/sile-typesetter/sile/blob/1a86520d198d7709966e1fd1bfd8dfaceda86393/typesetters/base.lua#L571-L574

I then tried:
```
\begin{document}
\vbox{}
\skip[height=30pt]
\center{hello world}
\end{document}
```
Also putting the `\skip` into the `\vbox` did not work.

I read through the source trying to grasp how the data / content flows through the typesetter (and where exactly my whitespace is dropped), but could not understand it.
Is there a high-level overview how the typesetter / data model in SILE works? The in-source comments were not sufficient for me.

Also, why does `\vbox{content goes here}` not  produce visible output?

**Comment by Omikhleia** [comment, marked as answer]
Quick answer:
Using an `\hbox{}` instead of a `\vbox{}` does the trick (more or less).

Longer delails:
`\vbox{}` doesn't output anything (it collates it potential content, wraps it via a temporary typesetter state in a vbox node (actually a vbox list), and returns it for consumption by other routines (... which might make use of it in Lua - e.g. it's used in the footnotes logic -, but it won't be of any use in a SIL document). These things are really arcane low-level constructs.

**Reply by Omikhleia** [reply]
BTW the in-code comment mentioning an empty vbox is not wrong _per se_. At this point, the typesetter _is_ processing a list of vboxes (lines are vboxes, possibly containing hboxes and other horizontal contents).

**Reply by raphCode** [reply]
Thank you, `\hbox` works!

In generally I am interested in the inner workings of the tools I use, and thus my desire to understand SILE better still stands.
I feel a glossary of the terms used in the code (like you just wrote about vboxes) and some explanation for important functions would go a long way.
Especially on a typesetting system that is designed to be hacked.

If I can figure out enough of it I may write something down eventually.

**Reply by Omikhleia** [reply]
Your are welcome!
The problem would be maintaining such a documentation - lots of things may still change / evolve. In-code comments are also lacking in many places in the code (IMHO, the maintainers could disagree - but sometimes the intent behind some tricky code is far from obvious and could be better explained).
